### DISCUSSION TITLE
Weird behavior with windows opening on the wrong workspace after start

### DISCUSSION THREAD

**Discussion by philippludwig** [i3/i3#5888] [Q&A]
I have one minor issue with using i3, where I am not sure if it happens because I am using it wrong. What I want is that applications open on the currently active workspace, and I believed this was the case in the past (been using i3 for at least 8 years now). However, for some time, its like this:

1. I start an i3 session
2. Workspace "1" is active
3. I open some application
4. I go to another workspace, e.g. "3" and open some application (e.g. xfce4-terminal)
5. The application opens on workspace 1
6. I need to go to workspace 1 and manually move the window to workspace 3
7. From now on, for the rest of the session, applications open on the currently active workspace.

So in summary, after starting i3, I have to manually move an already open window to another workspace; if I don’t do that, all applications open on workspace 1, even if I am on a different workspace. What I would like to have is that applications always open on the current workspace, right after starting the session.

I am on version 4.22, but I am currently compiling 4.23 so maybe that already changed. Anyway, any insight on this will be appreciated.

**Comment by orestisfl** [comment]
This is because of the [startup protocol](https://www.freedesktop.org/wiki/Software/startup-notification/). Using `exec --no-startup-id` should ignore the protocol.

E.g. test with:
```
i3-msg 'exec --no-startup-id <executable>, workspace 3'
```

**Reply by philippludwig** [reply]
I see. However, I use rofi to start my applications; rofi is started via sxhkd.
